src/main/java/at/hannibal2/skyhanni/features/misc/halloweenlobbywaypoints/BasketWaypoints.kt

```diff
@@ -0,0 +1,61 @@
+package at.hannibal2.skyhanni.features.misc.halloweenlobbywaypoints
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.ScoreboardData
+import at.hannibal2.skyhanni.test.GriffinUtils.drawWaypointFilled
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RenderUtils.drawDynamicText
+import net.minecraft.client.Minecraft
+import net.minecraftforge.client.event.RenderWorldLastEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class BasketWaypoints {
+    private val config get() = SkyHanniMod.feature.misc.halloweenBasket
+    private var waypoint: LorenzVec? = null
+    private var waypointName: String? = null
+
+    @SubscribeEvent
+    fun onRenderWorld(event: RenderWorldLastEvent) {
+        if (LorenzUtils.inSkyBlock || !LorenzUtils.onHypixel) return
+
+        if (!chechScoreboardHalloweenSpecific()) return
+
+        if (config.allWaypoints) {
+            for (basket in Basket.entries) {
+                event.drawWaypointFilled(basket.waypoint, LorenzColor.GOLD.toColor())
+                event.drawDynamicText(basket.waypoint, "§6" + basket.basketName, 1.5)
+            }
+        }
+
+        if (config.allEntranceWaypoints) {
+            for (basketEntrance in BasketEntrances.entries) {
+                event.drawWaypointFilled(basketEntrance.waypoint, LorenzColor.YELLOW.toColor())
+                event.drawDynamicText(basketEntrance.waypoint, "§e" + basketEntrance.basketEntranceName, 1.5)
+            }
+            return
+        }
+
+        if (LorenzUtils.skyBlockArea == "?") return
+
+        waypoint?.let {
+            event.drawWaypointFilled(it, LorenzColor.GOLD.toColor())
+            event.drawDynamicText(it, "§6" + waypointName!!, 1.5)
+        }
+    }
+
+    private fun chechScoreboardHalloweenSpecific(): Boolean {
```

Comment:
Please don't make this "heavy function" (compared to a boolean access) in a render call. Instead, use a tick event every second or so and store a variable "isHalloweenScoreboard"